libminifi/src/core/state/nodes/SupportedOperations.cpp

```diff
@@ -62,11 +63,27 @@ void SupportedOperations::addProperty(SerializedResponseNode& properties, const
   properties.children.push_back(operand_node);
 }
 
+std::vector<std::string> SupportedOperations::getSensitiveProperties() const {
+  std::vector<std::string> sensitive_properties;
+  sensitive_properties.assign(Configuration::DEFAULT_SENSITIVE_PROPERTIES.begin(), Configuration::DEFAULT_SENSITIVE_PROPERTIES.end());
+  if (configuration_reader_) {
+    const auto additional_sensitive_props_list = configuration_reader_(Configuration::nifi_sensitive_props_additional_keys);
+    if (additional_sensitive_props_list) {
+      std::vector<std::string> additional_sensitive_properties = utils::StringUtils::split(*additional_sensitive_props_list, ",");
+      return Configuration::mergeProperties(sensitive_properties, additional_sensitive_properties);
+    }
+  }
+  return sensitive_properties;
+}
```

Comment:
it seems to me, that there are three separate instances of building up the sensitive properties list, could we somehow unify these?

Reply:
I was thinking about this, when I checked how these properties are used in the encrypt-config. There was one major difference though that the encrypt-config reads these properties from a ConfigFile object which is a PropertiesFile representation and not from a whereas we only have a function object available to read a single property from a Configuration object in the SupportedProperties.

As for the tests we also have a similar problem where we only pass a function object to the SupportedOperations to reach the configuration properties while in the tests we have the actual Configuration object available, so it's not that easy to generalize it, but I was also worried about to use the same code that we are testing in the test as well.

Reply:
the least common denominator seems to be the function object, we could go for a static `Configuration::getSensitiveProperties(std::function<std::string(const std::string)> reader)`

Reply:
That's a fair point, extracted the function in 17c2db229f70f3453489cc9e2a7a96a65422da15